gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -0,0 +1,1034 @@
+/******************************************************************************
+ *
+ * Project:  FlatGeobuf driver
+ * Purpose:  Implements OGRFlatGeobufLayer class
+ * Author:   Björn Harrtell <bjorn at wololo dot org>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018-2019, Björn Harrtell <bjorn at wololo dot org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "ogrsf_frmts.h"
+#include "cpl_vsi_virtual.h"
+#include "cpl_conv.h"
+#include "cpl_json.h"
+#include "cpl_http.h"
+#include "ogr_p.h"
+
+#include "ogr_flatgeobuf.h"
+
+using namespace flatbuffers;
+using namespace FlatGeobuf;
+
+OGRFlatGeobufLayer::OGRFlatGeobufLayer(const Header *poHeader, GByte *headerBuf, const char *pszFilename, uint64_t offset)
+{
+    CPLDebug("FlatGeobuf", "offset: %lu", static_cast<long unsigned int>(offset));
+
+    m_poHeader = poHeader;
+    m_headerBuf = headerBuf;
+
+    CPLAssert(poHeader);
+    CPLAssert(pszFilename);
+
+    if (pszFilename)
+        m_osFilename = pszFilename;
+    m_offsetInit = offset;
+    m_offset = offset;
+    m_create = false;
+
+    m_featuresCount = m_poHeader->features_count();
+    CPLDebug("FlatGeobuf", "m_featuresCount: %lu", static_cast<long unsigned int>(m_featuresCount));
+    m_geometryType = m_poHeader->geometry_type();
+    m_hasZ = m_poHeader->hasZ();
+    m_hasM = m_poHeader->hasM();
+    m_hasT = m_poHeader->hasT();
+
+    CPLDebug("FlatGeobuf", "m_hasZ: %d", m_hasZ);
+    CPLDebug("FlatGeobuf", "m_hasM: %d", m_hasM);
+    CPLDebug("FlatGeobuf", "m_hasT: %d", m_hasT);
+
+    auto crs = m_poHeader->crs();
+    if (crs != nullptr) {
+        m_poSRS = new OGRSpatialReference();
+        auto org = crs->org();
+        auto code = crs->code();
+        auto wkt = crs->wkt();
+        m_poSRS->SetAuthority(nullptr, org->c_str(), code);
```

Comment:
this is likely useless. A SRS with just an authority node is invalid

Reply:
Hmm ok, the idea was to handle the case where org is something other than EPSG and no WKT is supplied and as such cannot be imported by the logic that follows.